src/vector/index.html

```diff
@@ -68,4 +68,10 @@
     <!-- let CSS themes pass constants to the app -->
     <div id="mx_theme_accentColor"></div><div id="mx_theme_secondaryAccentColor"></div><div id="mx_theme_tertiaryAccentColor"></div>
   </body>
+
+  <script>
```

Comment:
will this not break due to CSP rules against inline-javascript?

Reply:
I wasnt aware CSP rules were in place, would you prefer a new file or is there a convenient place in the existing js to put this?

Reply:
it should probably be shoved through webpack like the indexeddb-worker.

Reply:
The CSP hasn't yet been moved into a meta tag of the app, so it's bit hard to discover (it's only currently set server side for riot.im). At the moment, it includes:

```
default-src 'none'; 
style-src 'self' 'unsafe-inline';
script-src 'self' 'unsafe-eval' 'unsafe-inline' https://www.recaptcha.net https://www.gstatic.com; 
img-src * blob: data:; 
connect-src *; 
font-src 'self'; 
media-src * blob: data:; 
child-src * blob: data:;
form-action 'self'; 
object-src 'self'; 
manifest-src 'self'
```

`unsafe-inline` would seem to say that inline script tags are allowed, and indeed we seem to already use one as with the IDB worker snippet above.

Given all of that, this seems okay where it is (at least CSP-wise).

@daleharvey, I think it would be a little tidier if we moved this up by the other script blocks (maybe line 44, so it runs last, since it's just a shim).

Reply:
(except that we also shouldn't be hitting `/` because pathing is a thing)

Reply:
Ah yes, we don't want to take over the entire domain, as the app is commonly hosted of a path like `https://riot.im/app`, so perhaps adding `{ scope: './' }` to the `register` call is enough for that.

Reply:
Service workers cant take over urls below the path they are requested on (`scope` can only limit them further), `https://riot.im/app/sw.js` will take control of requests under `https://riot.im/app/`, `https://localhost:8080/sw.js` will take control of requests under `https://localhost:8080/`

Reply:
Also just realised what @turt2live meant sorry, using `/sw.js` as the path would have broken it under `/app/`, so fixed that, the service worker scope should still do the right thing